modules/meta/arrow-meta-prototype/testing-plugin/src/main/kotlin/arrow/meta/plugin/testing/Compilation.kt

```diff
@@ -17,12 +19,16 @@ enum class CompilationResult {
 data class CompilationData(
   val sourceFileName: String,
   val sourceContent: String,
-  val generatedFileContent: String,
+  val generatedFileContent: Optional<String>,
   val generatedClasses: ArrayList<String>,
-  val compilationResult: CompilationResult
+  val compilationStatus: CompilationStatus
 )
 
-fun testCompilation(compilationData: CompilationData): Unit {
+data class CompilationResult(
+  val classesDirectory: String
+)
+
+fun testCompilation(compilationData: CompilationData): Optional<CompilationResult> {
```

Comment:
Use Kotlin nullable types instead of Optional

Reply:
Thanks! Done! Clearer code now :tada: